### DISCUSSION TITLE
Why I believe this could be an unofficial replacement /alternative for rust

### DISCUSSION THREAD

**Discussion by YalekW5435** [carbon-language/carbon-lang#3588] [General]
There's a couple reasons I mention this and it's because if you combine oxygen and iron together, you get rust.. and based off of the looks of the language it feels Rusty and it feels like a rusted out version of c++ even though yes it has its ups and downs, it is not object oriented and the documentation never mentions of such.  And Rust as an element is not an actual physical object, it's more or less scraped off steel.
However carbon and iron make *steel*.
Not many people usually talk about a chemical context or atomic contexts when it comes down to programming code.
There's a bunch of reasons why I say these things.  

When I look at rust, this language is just extremely syntactically different and read differently from my vantage point then carbon is.  Like. Where did "fn or function" come from? (Fn makes use here in carbon through).
Are you trying to put javascript in here?  Any c based languages after c#, java based, etc. also use var.
What the heck is a var?  It's just a generic variable that I don't know what's going to happen to it.  IN MY EYES THAT IS  ##ABSOLUTELY THE MOST TRAGIC THING THAT CAN HAPPEN TO A COMPUTER IS NOT KNOWING WHAT SOMETHING IN THE CODE IS DOING!! (Caps intended).  

Another thing about Rust is that programming languages community tends to try to force that language into your head by having you ignore the other languages and superiorizing theirs. By saying "rust is the best language don't ever use any others."

Now, I'm not sure if this is true everywhere, but I'm pretty sure my friends have observed this exact same attitude from those very same people on rust community forums and chats.

Now yes the carbon language needs to be worked on, still has a bunch of issues that need to be resolved, but I believe this could be the second if not topping language that would go ahead of rust one day...
As this looks pretty similar to it but a little bit farther away from it..

Remember when I mentioned chemical compounds and you know steel carbon oxygen all these other atoms and molecular structures? you have to look at that from that context in order to understand what I'm talking about here.
It's the very same thing that code is formed and that's why carbon is now going to exist alongside rust, possibly as the two biggest competitors I've ever seen.

**Comment by geoffromer** [comment]
Just to be clear, we [do not intend or expect](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/faq.md#if-you-can-use-rust-ignore-carbon) Carbon to be a replacement for Rust.

**Comment by YalekW5435** [comment]
Yeah..
I just think it'll be more of an alternative to rust, rather than an actual
replacement.

**Comment by YalekW5435** [comment]
I only said replacement because it could be a huge competitor to rust and
could be a huge gateway to an alternative.
I'm not sure if this is the proper explanation if somebody can clarify it
for me please do..

**Comment by YalekW5435** [comment]
There was one language I was actually going to propose but never got around
to doing.
It was called "C=".
Now I'm not saying that carbon should be changed to that, because that
would read the name of this language to begin with.
Now the actual description of this language what describe the very same
thing that C= would do.
The biggest problem I see here is that C equals is heavily based off of the
actual C++ library and would probably be a little different looking and
would look a little bit like carbon and rust too.. and for me to form
compilers will be a disaster cuz I am not into that field of study yet. 😂.

**Comment by felixf4xu** [comment]
As long as there is a morden drop-in replacement for c++ compiler, rust will not be the first option.

**Comment by WilliamYe79** [comment]
My main working language is Java, and I am very comfortable with it in at least 80% scenarios. However, it is the case that I still need to use Python/Go/Rust/C to do something, which is hard or impractical to be implemented with Java.
C++ was the 1st OOP I learnt. Although I didn't dig very deep in it, thanks to my good memory, I am still familiar with its syntax.
Last year, I bought a book of Rust and did some practice. I can say: Rust is quite innovative and it is probably the best option to develop an Operating System. However, for me, the syntax is really over complicated and not friendly or ergonomic at all. I am really curious why they designed the syntax of Rust in such a way.
I also learnt a bit Zig, which is far more developer friendly than Rust. However, it still seems a bit unnatural to me.
Today, when I read the docs of Carbon, I found that I really love its syntax. It's so family and compliant with the intuitive of an OOP programmer.
Looking forward to any progress of Carbon.

**Reply by richard-hp** [reply]
I don't think it's as much that the Rust *syntax* is difficult to understand.

It's more that Rust uses a fundamentally different model of computation that you have to wrap your head around.  The syntax are just symbols that represent the ideas underlying how you model problems and represent those in code.

If you approach Rust thinking it's like any other program language, you will not get very far with it.  You pretty much have to read the entire Rust book from cover to cover (it's not that long), and actually spend some time learning and understanding the different principles behind it, the design decisions etc.

Most other languages are basically the same but with different syntax, and maybe some different quirks about how you do certain things.  But once you've learned say, JavaScript.  You can go from there and learn Python, and a lot of it is "how do I do this thing I already know how to do in JS, in X language".

You are essentially, when learning these related languages, just learning how to do the same kind of things in a new syntax.  Rust is not like this.  You cannot build software in the same way in JS as you can in Rust.  It is built on different principles and if you don't wrap your head around those then the syntax will not make much sense.

**Comment by YalekW5435** [comment]
Good rule of thumb.
